src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -205,8 +205,8 @@ class NeighborCollection
   }
 
 private:
-  std::int8_t num_nbrs[MAX_VERTS];
-  VertexNbrs nbrs[MAX_VERTS];
+  std::int8_t num_nbrs[MAX_VERTS] {};
+  VertexNbrs nbrs[MAX_VERTS] {};
```

Comment:
The cuda test case passes when I revert this default member initalizer change. I am not sure why this causes problem on device...

Reply:
!!!  Well, thank you for checking on that, Brian.  That's very odd.

Reply:
Great catch @bmhan12 -- looks like uninitialized memory